src/libs/tools/src/plugindatabase.cpp

```diff
@@ -367,5 +364,39 @@ std::string MockPluginDatabase::lookupInfo (PluginSpec const & spec, std::string
 
 	return "";
 }
+
+/**
+* @brief reference checkconf implementation for the unit tests.
+* 
+* Checks for the key "a" to be set to "abc" and fails if it can not find it.
+* Checks for the key "b" and appends it to the configuration if it can not find it.
+*/
+static int checkconfMock (ckdb::Key * errorKey, ckdb::KeySet * conf)
+{
+	ckdb::Key * k;
+
+	k = ckdb::ksLookupByName (conf, "user/a", 0);
+	if (!k || strncmp (ckdb::keyString (k), "abc", 3))
+	{
+		return -1; // invalid config, could not fix
+	}
+
+	k = ckdb::ksLookupByName (conf, "user/b", 0);
+	if (!k)
+	{
+		ckdb::ksAppendKey (conf, ckdb::keyNew ("user/b", 0));
+		return 1; // changed config, OK
+	}
+	return 0; // did not change config, OK
```

Comment:
I would split that up into multiple mocks, maybe the problem is in the test case. Test cases should be as simple as possible.